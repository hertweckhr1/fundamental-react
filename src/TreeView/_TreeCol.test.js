import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TreeCol from './_TreeCol';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<TreeCol />', () => {
    const treeCol = (
        <TreeCol>Grandchild 1</TreeCol>
    );

    test('create tree col component', () => {
        // tree col
        let component = renderer.create(treeCol);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TreeCol component', () => {
            const element = mount(
                <TreeCol
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree__col').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
