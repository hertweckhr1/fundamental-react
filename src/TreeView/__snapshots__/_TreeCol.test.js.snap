// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeCol /> create tree col component 1`] = `
<div
  className="fd-tree__col"
>
  Grandchild 1
</div>
`;
